wordpress-mobile/WordPress-Android
# [Dependency Updates] Remove Room Dependency

Parent #17564

This PR is about removing the (hopefully) no longer necessary `Room` dependency from this project. This set of `Room` related test dependencies, which were only targeting UI test, were added as part of the below PR and only in order to fix some connected tests from failing due to classpath errors, see below:

PR: [Add room libraries to fix connected tests #14974](https://github.com/wordpress-mobile/WordPress-Android/pull/14974) Cc @planarvoid

```
 e: Supertypes of the following classes cannot be resolved. Please make
sure you have the required dependencies in the classpath:
    class org.wordpress.android.fluxc.persistence.WPAndroidDatabase,
     unresolved supertypes: androidx.room.RoomDatabase
```

However, these kind of errors were also appearing on unit tests at that point. Still, after a while and even though the above fix PR only targeted Android tests, it appeared that the unit tests were also working as expected, at least for a while, and at least for most of our engineers. @mkevins was an exception to that as he reported that he experienced this issue again 10 months ago. But, even then, by just rebuilding the project, this issue seems to resolve itself.

As such, with this commit, I recommend trying our removing this `Room` dependency from the project and then monitor this change for a few days/weeks/months to see if this problems reappears. Or else, whether this problem has been now automatically resolved by itself, due to maybe an IDE, AGP or any other update that might have potentially fix this issue automatically for us.

With this change, the project will also become more lightweight as this will mean maintaining one less dependency overall.

-----

PS: Also, [WCAndroid](https://github.com/woocommerce/woocommerce-android), which also using [FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android) and thus utilizing `Room` the same way this project does, never applied the same fix, that is, to add room libraries to fix any connected or unit test. This means that this extra `Room` dependency is unnecessary on this project as well and should/could be potentially removed for good.

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could checkout this branch locally and try to run some Android and unit test, both via AS and CLI, and verify correctness, which is, that the tests are not failing to compile with some kind of a `Room` related classpath error such as the aforementioned one above.

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:
